temboardui/json_encoder.py

```diff
@@ -1,11 +1,42 @@
-import json
 from sqlalchemy.ext.declarative import DeclarativeMeta
+from json import JSONEncoder as BaseJSONEncoder
+from datetime import datetime
+from decimal import Decimal
+
+
+class JSONEncoder(BaseJSONEncoder):
+    """
+    JSONEncoder used throughout the application.
+    Handle Decimal, datetime and JSONizable objects.
+    """
+    def default(self, obj):
+        if isinstance(obj, Decimal):
+            return float(obj)
+        if isinstance(obj, datetime):
+            return obj.strftime("%Y-%m-%d %H:%M:%S%z")
+        if isinstance(obj, JSONizable):
+            return obj.to_json()
+        return BaseJSONEncoder.default(self, obj)
+
+
+class JSONizable(object):
+    """
+    Base class for an object which is serializable to JSON.
+    """
+    def to_json(self):
+        """
+        Serialize the object to JSON
+        Returns:
+            an object which can be encoded by the BaseJSONEncoder.
+        """
+        return dict(((key, val) for key, val in self.__dict__.items()
+                    if not key.startswith("_")))
```

Comment:
This class does not appear to be used anywhere in the PR. Or maybe I missed something?

Reply:
You have the following statement in the JSONEndocer class (same file):
`if isinstance(obj, JSONizable):`

Reply:
Yes, I noticed that. But it seems that there's no class inheriting from JSONizable at this point so this code will never apply, if I understand correctly.

Reply:
My bad.